TestAssets/TestPackages/dotnet-dependency-context-test/project.json

```diff
@@ -1,5 +1,5 @@
 {
-  "version": "1.0.0-preview1-*",
+  "version": "1.0.0-preview2-*",
```

Comment:
I am hitting the following exception in Pack when I try to remove preview1

>[EXEC       >] [....] [00:05:14.184] D:\github\cli\artifacts\win10-x64\stage2\dotnet.exe "pack" "D:\github\cli\TestAssets\TestPackages\dotnet-dependency-context-test\project.json" "--no-build" "--build-base-path" "D:\github\cli\artifacts\win10-x64\tests\packagesBuild" "--output" "D:\github\cli\artifacts\win10-x64\tests\packages" "--version-suffix" "002703"
D:\github\cli\TestAssets\TestPackages\dotnet-dependency-context-test\project.json(0,0): Error: Microsoft.DotNet.ProjectModel.FileFormatException: '1.0.0-002703' is not a valid version string.
Parameter name: value ---> Microsoft.DotNet.ProjectModel.FileFormatException: '1.0.0-002703' is not a valid version string.
Parameter name: value ---> System.ArgumentException: '1.0.0-002703' is not a valid version string.
Parameter name: value
   at NuGet.Versioning.NuGetVersion.Parse(String value)
   at NuGet.Versioning.NuGetVersion..ctor(String version)
   at Microsoft.DotNet.ProjectModel.ProjectReader.SpecifySnapshot(String version, String snapshotValue)
   at Microsoft.DotNet.ProjectModel.ProjectReader.ReadProject(Stream stream, String projectName, String projectPath, ProjectReaderSettings settings)
   --- End of inner exception stack trace ---
   at Microsoft.DotNet.ProjectModel.ProjectReader.ReadProject(Stream stream, String projectName, String projectPath, ProjectReaderSettings settings)
   at Microsoft.DotNet.ProjectModel.ProjectReader.TryGetProject(String path, Project& project, ProjectReaderSettings settings)
   --- End of inner exception stack trace ---
   at Microsoft.DotNet.ProjectModel.ProjectReader.TryGetProject(String path, Project& project, ProjectReaderSettings settings)
   at Microsoft.DotNet.ProjectModel.Workspace.AddProjectEntry(String projectDirectory, FileModelEntry`1 currentEntry)
   at Microsoft.DotNet.ProjectModel.Workspace.<GetProjectCore>b__11_0(String key)
   at System.Collections.Concurrent.ConcurrentDictionary`2.AddOrUpdate(TKey key, Func`2 addValueFactory, Func`3 updateValueFactory)
   at Microsoft.DotNet.ProjectModel.Workspace.GetProjectCore(String projectDirectory)
   at Microsoft.DotNet.ProjectModel.Workspace.AddProjectContextEntry(String projectDirectory, ProjectContextCollection currentEntry)
   at Microsoft.DotNet.ProjectModel.Workspace.<GetProjectContextCollection>b__8_0(String key)
   at System.Collections.Concurrent.ConcurrentDictionary`2.AddOrUpdate(TKey key, Func`2 addValueFactory, Func`3 updateValueFactory)
   at Microsoft.DotNet.ProjectModel.Workspace.GetProjectContextCollection(String projectPath)
   at Microsoft.DotNet.Tools.Compiler.PackCommand.<>c__DisplayClass0_0.<Run>b__0()
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.DotNet.Tools.Compiler.PackCommand.Run(String[] args)